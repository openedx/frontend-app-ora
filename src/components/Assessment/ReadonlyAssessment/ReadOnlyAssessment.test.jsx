import { shallow } from '@edx/react-unit-test-utils';

import ReadOnlyAssessment from './ReadOnlyAssessment';

jest.mock('./CollapsibleAssessment', () => 'CollapsibleAssessment');
jest.mock('./AssessmentCriteria', () => 'AssessmentCriteria');

describe('<ReadOnlyAssessment />', () => {
  const props = {
    assessment: {
      abc: 'def',
    },
    step: 'Step',
    stepScore: {
      earned: 5,
      total: 10,
    },
  };

  it('render with assessment', () => {
    const wrapper = shallow(<ReadOnlyAssessment {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('AssessmentCriteria').length).toBe(1);
  });

  it('render with assessments', () => {
    const assessments = [
      {
        abc: 'def',
      },
      {
        ghi: 'jkl',
      },
    ];
    const wrapper = shallow(<ReadOnlyAssessment {...props} assessments={assessments} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('AssessmentCriteria').length).toBe(2);
  });

  it('renders without props', () => {
    const wrapper = shallow(<ReadOnlyAssessment />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
