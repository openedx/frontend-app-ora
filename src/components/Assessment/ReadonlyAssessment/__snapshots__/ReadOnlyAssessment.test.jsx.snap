// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadOnlyAssessment /> render with assessment 1`] = `
<CollapsibleAssessment
  defaultOpen={false}
  stepLabel={null}
  stepScore={
    Object {
      "earned": 5,
      "total": 10,
    }
  }
>
  <AssessmentCriteria
    abc="def"
    stepLabel={null}
  />
</CollapsibleAssessment>
`;

exports[`<ReadOnlyAssessment /> render with assessments 1`] = `
<div
  className="my-2"
  key="Step"
>
  <CollapsibleAssessment
    defaultOpen={false}
    stepLabel={null}
    stepScore={
      Object {
        "earned": 5,
        "total": 10,
      }
    }
  >
    <Fragment
      key="some_uuid"
    >
      <p
        className="mb-0"
      >
         
        1
        : 
      </p>
      <AssessmentCriteria
        abc="def"
        stepLabel={null}
      />
      <hr
        className="my-4"
      />
    </Fragment>
    <Fragment
      key="some_uuid"
    >
      <p
        className="mb-0"
      >
         
        2
        : 
      </p>
      <AssessmentCriteria
        ghi="jkl"
        stepLabel={null}
      />
      <hr
        className="my-4"
      />
    </Fragment>
  </CollapsibleAssessment>
</div>
`;

exports[`<ReadOnlyAssessment /> renders without props 1`] = `
<CollapsibleAssessment
  defaultOpen={false}
  stepLabel={null}
  stepScore={null}
>
  <AssessmentCriteria
    stepLabel={null}
  />
</CollapsibleAssessment>
`;
